File: 基于最近邻超圆盘分类的CWRU轴承数据集故障识别/apply_fft.py
import numpy as np
from scipy.fftpack import fft

def apply_fft(x, fs, num_samples):
    f = np.linspace(0.0, (fs/2.0), num_samples//2)
    freq_values = fft(x)
    freq_values = 2.0/num_samples * np.abs(freq_values[0:num_samples//2])
    return f, freq_values

File: 基于最近邻超圆盘分类的CWRU轴承数据集故障识别/apply_fft.m
function [f, freq_values] = apply_fft(x, fs, num_samples)
    f = linspace(0.0, (fs/2.0), num_samples/2);
    freq_values = fft(x);
    freq_values = 2.0/num_samples * abs(freq_values(1:num_samples/2));
end
